### PR TITLE
add support for the GetSecretValue restful API

### PR DESCRIPTION
**Pull request by wanghouqi**
this restful API will be exposed to public as:
 POST <ehsm_srv_address>/ehsm?Action=GetSecretValue

 Obtains a secret value.

 test: passed the python test

Signed-off-by: wanghouqi <wanghouqi@vip.qq.com>

### CODE REVIEW

File: ehsm_kms_service/secret_manager_apis.js [hunk 1]
```diff
@@ -265,9 +265,108 @@ const createSecret = async (res, appid, payload, DB) => {
...
+    try {
+        // get and check parameter
+        let secretName = getParam_String(payload, 'secretName')
+        let versionId = payload['versionId']
+        let selector;
+        if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+        if (versionId != undefined && typeof (versionId) !== 'number') {
```
Comment (yang8621) [inline ehsm_kms_service/secret_manager_apis.js:279]: if the version is not passed, will return the secrect of the latest version?

File: ehsm_kms_service/secret_manager_apis.js [hunk 2]
```diff
@@ -265,9 +265,108 @@ const createSecret = async (res, appid, payload, DB) => {
...
+            res.send(_result(400, 'versionId must be a integer.'))
+            return
+        }
+
+        // build selector for search CouchDB, if has versionId use it find, 
+        // if not use versionStage = SM_SECRET_VERSION_STAGE_CURRENT
+        if (versionId) {
```
Comment (yang8621) [inline ehsm_kms_service/secret_manager_apis.js:286]: How about the version is 0 here?

File: ehsm_kms_service/secret_manager_apis.js [hunk 3]
```diff
@@ -265,9 +265,113 @@ const createSecret = async (res, appid, payload, DB) => {
...
+        const query = {
+            selector,
+            fields: ['secretData', 'versionId', 'createTime'],
+            limit: 1,
+        }
+        let query_result = await DB.partitionedFind('secret_version_data', query)
+
+        // build result
+        let result = {}
+        if (query_result.docs.length > 0) {
```
Comment (HawieLong) [inline ehsm_kms_service/secret_manager_apis.js:317]: where is this if's else?

File: test/test_secret_manager_with_cli.py [hunk 4]
```diff
@@ -26,10 +26,18 @@ def test_secret_manager(base_url, headers):
...
     key1 = createkey.createkey(base_url, "EH_AES_GCM_128", "EH_INTERNAL_KEY")
 
+    # createSecret with encryptionKeyId
     createSecret.createSecret(base_url, "secretData1", "secret001", key1, "mysecret", "30h")
-
+    # createSecret use default CMK
     createSecret.createSecret(base_url, "secretData2", "secret002", None, "mysecret", "20d")
 
+    # getSecretValue by current version
+    secretData1 = getSecretValue.getSecretValue(base_url, "secret001", None)
```
Comment (syan10) [inline test/test_secret_manager_with_cli.py:35]: can we only pass 2 parameters, the default is None.

File: ehsm_kms_service/secret_manager_apis.js (hunk 2 above)
Comment (yang8621) [reply ehsm_kms_service/secret_manager_apis.js:304]: will it go to Line 311 ?

**Comment by wanghouqi** [comment]
Move to https://github.com/intel/ehsm/pull/155

**Pull request closed without merging**